cmd/rpcdaemon/commands/trace_filtering.go

```diff
@@ -366,6 +368,12 @@ func (api *TraceAPIImpl) Filter(ctx context.Context, req TraceFilterRequest, str
 
 		blockHash := block.Hash()
 		blockNumber := block.NumberU64()
+		if !isPos {
+			isPos, err = rawdb.IsPosBlock(dbtx, blockHash)
```

Comment:
Can you just look at the block header (difficulty) and see if it is POS, otherwise this will hit the DB for every block before POS